### ISSUE TITLE
`strides` missing methods for various  AbstractArray subtypes 

### ISSUE THREAD

**Issue by musm** [JuliaLang/julia#29705]
Seems like some simple methods are missing for `strides` function
```
julia> strides(randn(5,5)')
ERROR: MethodError: no method matching strides(::LinearAlgebra.Adjoint{Float64,Array{Float64,2}})

```

similarily
```
julia> strides(transpose(randn(5,5))
```
and also other basic subtypes of `AbstractArray` such as `Diagonal` etc.

**Comment by korbinian90** [comment]
Is this still worked on? No method is available for BitArrays in Julia 1.4.1

```
julia> strides([1,0] .> 0)
ERROR: MethodError: no method matching strides(::BitArray{1})
```

Also `NIVolume`, the `AbstractArray` subtype from `NIfTI.jl` doesn't work with strides.

**Comment by mbauman** [comment]
In Julia, we've used the stride terminology to talk about storage locations.  BitArrays don't have strided storage locations, so they'll never support `strides`.

**Comment by korbinian90** [comment]
Thanks for your answer!
As I was looking for a different concept, I should name my function something like `linear_neighbor_offsets`.
And probably I should try to use the 'safe' way with `LinearIndices`.

**Comment by mbauman** [comment]
This particular example is now fixed in #29135.  I think we can roll the rest into the bigger #30432.
